src/QueuePerThreadPool.c

```diff
@@ -268,29 +272,43 @@ static void claim_work(QPTPoolThreadData_t *tw) {
     pthread_mutex_unlock(&tw->claimed_mutex);
 }
 
-#if defined(DEBUG) && defined (QPTPOOL_QUEUE_SIZE)
 static void dump_queue_size_stats(QPTPool_t *ctx, QPTPoolThreadData_t *tw) {
+    #if defined(DEBUG) && defined (QPTPOOL_QUEUE_SIZE)
     pthread_mutex_lock(&ctx->mutex);
     pthread_mutex_lock(&print_mutex);
     tw->waiting.size = tw->claimed.size;
 
-    struct timespec now;
-    clock_gettime(CLOCK_MONOTONIC, &now);
-    fprintf(stderr, "qptpool_size %" PRIu64 " ", since_epoch(&now) - epoch);
+    char buf[4096];
+    char *p = buf;
 
     size_t sum = 0;
     for(size_t i = 0; i < ctx->nthreads; i++) {
-        fprintf(stderr, "%zu ", ctx->data[i].waiting.size);
+        p += snprintf(p, sizeof(buf), "%zu ", ctx->data[i].waiting.size);
         sum += ctx->data[i].waiting.size;
-        fprintf(stderr, "%zu ", ctx->data[i].deferred.size);
+        p += snprintf(p, sizeof(buf), "%zu ", ctx->data[i].deferred.size);
         sum += ctx->data[i].deferred.size;
     }
-    fprintf(stderr, "%zu\n", sum);
+
+    if (sum > ctx->queue_size_highwater) {
```

Comment:
Maybe check `ctx->incomplete` instead of generating the string only to throw it away.

Reply:
Can you elaborate on what you mean here -- are you saying to basically wrap the whole function body in `if (ctx->incomplete) { ... }`?

Reply:
Yeah. The `ctx->mutex` is locked already.

Reply:
Done.